clap_complete/src/dynamic/completer.rs

```diff
@@ -71,8 +71,53 @@ pub fn complete(
         }
 
         if is_escaped {
-            pos_index += 1;
-            state = ParseState::Pos(pos_index);
+            match state {
+                ParseState::ValueDone => {
+                    // Begin parsing a new positional argument after `--`.
+                    let pos_arg = current_cmd
+                        .get_positionals()
+                        .find(|p| p.get_index() == Some(pos_index));
+                    match pos_arg.map(|arg| arg.get_num_args()) {
+                        Some(Some(range)) => {
+                            // HACK: Assuming `num_args` is fixed.
+                            let max = range.max_values();
+                            debug!("complete::is_escape: max={:?}", max);
+                            if max > 1 {
+                                state = ParseState::Pos(1);
+                            } else {
+                                state = ParseState::Pos(1);
+                                pos_index += 1;
+                            }
+                        }
+                        _ => {
+                            state = ParseState::ValueDone;
+                            pos_index += 1;
+                        }
+                    }
+                }
+                ParseState::Pos(num_arg) => {
+                    // HACK: Assuming `num_args` is fixed.
+                    let pos_arg = current_cmd
+                        .get_positionals()
+                        .find(|p| p.get_index() == Some(pos_index));
+                    let count = match pos_arg.map(|arg| arg.get_num_args()) {
+                        Some(Some(range)) => range.max_values(),
+                        _ => 1,
+                    };
+                    if num_arg + 1 < count {
+                        state = ParseState::Pos(num_arg + 1);
+                    } else {
+                        state = ParseState::Pos(1);
+                        pos_index += 1;
+                    }
+                }
+                ParseState::Opt(..) => {
+                    return Err(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        "Parsing error in native completion",
+                    ));
```

Comment:
Unsure if errorring is the right thing to do...

Reply:
Logically, this branch will not be entered, but it must be matched. If there is a better way to implement it, please tell me.

Reply:
`unreachable!("reason it won't be hit")`